docs/guide_slash_and_interactions.rst

```diff
@@ -0,0 +1,292 @@
+
+Arguments
+---------
+As shown in some of the above examples, we can amplify our slash commands with arguments.
+However with slash commands Discord allows us to do a few more things.
+Such as specifically select a channel that we'd like to use in our commands,
+we can do the same with roles and members.
+Let's take a look at how we can do that.
+
+.. code-block:: python
+
+    import discord
+
+    from redbot.core import commands, app_commands
+
+    class MyCog(commands.Cog):
+        def __init__(self, bot):
+            self.bot = bot
+
+        @app_commands.command()
+        @app_commands.describe(channel="The channel you want to use")
+        async def channel(self, interaction: discord.Interaction, channel: discord.TextChannel):
+            await interaction.response.send_message(f"Your channel is {channel.mention}", ephemeral=True)
+
+        @app_commands.command()
+        @app_commands.describe(role="The role you want to use")
+        async def role(self, interaction: discord.Interaction, role: discord.Role):
+            await interaction.response.send_message(f"Your role is {role.mention}", ephemeral=True)
+
+        @app_commands.command()
+        @app_commands.describe(member="The member you want to use")
+        async def member(self, interaction: discord.Interaction, member: discord.Member):
+            await interaction.response.send_message(f"Your member is {member.mention}", ephemeral=True)
+
+If you try out the channel command, you will see that it currently accepts any type of channel,
+however let's say we want to limit this to voice channels only.
+We can do so by adjusting our type hint to :class:`discord.VoiceChannel` instead of :class:`discord.TextChannel`.
+
+.. code-block:: python
+
+    import discord
+
+    from redbot.core import commands, app_commands
+
+    class MyCog(commands.Cog):
+        def __init__(self, bot):
+            self.bot = bot
+
+        @app_commands.command()
+        @app_commands.describe(channel="The channel you want to use")
+        async def channel(self, interaction: discord.Interaction, channel: discord.VoiceChannel):
+            await interaction.response.send_message(f"Your channel is {channel.mention}", ephemeral=True)
+
+With integer and float arguments, we can also specify a minimum and maximum value.
+This can also be done to strings to set a minimum and maximum length.
+These limits will be reflected within Discord when the user is filling out the command.
+
+.. code-block:: python
+
+    import discord
+
+    from redbot.core import commands, app_commands
+
+    class MyCog(commands.Cog):
+        def __init__(self, bot):
+            self.bot = bot
+
+        @app_commands.command()
+        @app_commands.describe(number="The number you want to use, max 10")
+        async def number(self, interaction: discord.Interaction, number: app_commands.Range[int, None, 10]):
+            await interaction.response.send_message(f"Your number is {number}", ephemeral=True)
```

Comment:
```suggestion
        @app_commands.command()
        @app_commands.describe(channel="The channel you want to mention")
        async def mentionchannel(self, interaction: discord.Interaction, channel: discord.TextChannel):
            await interaction.response.send_message(f"That channel is {channel.mention}", ephemeral=True)

        @app_commands.command()
        @app_commands.describe(role="The role you want to mention")
        async def mentionrole(self, interaction: discord.Interaction, role: discord.Role):
            await interaction.response.send_message(f"That role is {role.mention}", ephemeral=True)

        @app_commands.command()
        @app_commands.describe(member="The member you want to mention")
        async def mentionmember(self, interaction: discord.Interaction, member: discord.Member):
            await interaction.response.send_message(f"That member is {member.mention}", ephemeral=True)

If you try out the mentionchannel command, you will see that it currently accepts any type of channel,
however let's say we want to limit this to voice channels only.
We can do so by adjusting our type hint to :class:`discord.VoiceChannel` instead of :class:`discord.TextChannel`.

.. code-block:: python

    import discord

    from redbot.core import commands, app_commands

    class MyCog(commands.Cog):
        def __init__(self, bot):
            self.bot = bot

        @app_commands.command()
        @app_commands.describe(channel="The channel you want to mention")
        async def mentionchannel(self, interaction: discord.Interaction, channel: discord.VoiceChannel):
            await interaction.response.send_message(f"That channel is {channel.mention}", ephemeral=True)

With integer and float arguments, we can also specify a minimum and maximum value.
This can also be done to strings to set a minimum and maximum length.
These limits will be reflected within Discord when the user is filling out the command.

.. code-block:: python

    import discord

    from redbot.core import commands, app_commands

    class MyCog(commands.Cog):
        def __init__(self, bot):
            self.bot = bot

        @app_commands.command()
        @app_commands.describe(number="The number you want to say, max 10")
        async def saynumber(self, interaction: discord.Interaction, number: app_commands.Range[int, None, 10]):
            await interaction.response.send_message(f"Your number is {number}", ephemeral=True)
```